pkg/operator/staticpod/controller/missingstaticpod/missing_static_pod_controller.go

```diff
@@ -105,13 +120,23 @@ func (c *missingStaticPodController) sync(ctx context.Context, syncCtx factory.S
 		if err != nil {
 			return err
 		}
-		threshold := gracePeriod + 150*time.Second
+
+		isSNO, success, err := c.isSNODeployment()
```

Comment:
here `success` is `preconditionFulfilled`

```
if !preconditionFulfilled {
	klog.V(4).Infof("unable to determine the cluster topology, waiting for the infra informer to sync")
	return nil
}
```

Reply:
Done